src/logic/calculate.js

```diff
@@ -0,0 +1,20 @@
+import Operate from './operate';
+
+const Calculate = ({ total, next, operation }, buttonName) => {
+  const data = { total, next, operation };
+
+  if (buttonName === 'AC') {
+    data.total = 0;
```

Comment:
- [ ] [OPTIONAL] Instead of changing the value of data directly create a new object or clone and return that. It's safer to not modfy the original object.